awx_collection/plugins/modules/tower_inventory_source.py

```diff
@@ -340,6 +340,16 @@ def main():
             elif state == 'absent':
                 params['fail_on_missing'] = False
                 result = inventory_source.delete(**params)
+            elif state == 'updated':
+                try:
+                    params['inventory_source_id'] = inventory_source.get(name=params['name'])['id']
+                except (exc.NotFound) as excinfo:
+                    module.fail_json(
+                        msg='Failed to update inventory source,'
+                        'inventory not found: {0}'.format(excinfo),
+                        changed=False
+                    )
+                result = inventory_source.update(params['inventory_source_id'], wait=True)
```

Comment:
This should hit conflicts with this refactor PR, which is expected to land soon.

https://github.com/ansible/awx/pull/5747/files#diff-72cdc9e83a727f2e42f7f9703ff61e7eR357

Looks like the indentation will change, and then some other obvious syntax changes will be needed to how requests are made. Otherwise, the overall structure of what is being done here still seems compatible.